src/i40e_main.c

```diff
@@ -14849,11 +14849,12 @@ static int i40e_xdp_setup(struct i40e_vsi *vsi, struct bpf_prog *prog,
 
 	if (old_prog)
 		bpf_prog_put(old_prog);
-#ifdef HAVE_AF_XDP_ZC_SUPPORT
+
 	/* Kick start the NAPI context if there is an AF_XDP socket open
 	 * on that queue id. This so that receiving will start.
 	 */
 	if (need_reset && prog) {
+#ifdef HAVE_AF_XDP_ZC_SUPPORT
 		for (i = 0; i < vsi->num_queue_pairs; i++)
```

Comment:
```suggestion	
	if (need_reset && prog) {
#ifdef HAVE_AF_XDP_ZC_SUPPORT
        /* Kick start the NAPI context if there is an AF_XDP socket open
	     * on that queue id. This so that receiving will start.
	     */
		for (i = 0; i < vsi->num_queue_pairs; i++)
```